onnx/defs/reduction/defs.cc

```diff
@@ -33,6 +34,47 @@ False instead of True.)DOC";
         "T",
         OpSchema::high_precision_numeric_types(),
         "Constrain input and output types to high-precision numeric tensors.");
+    schema.TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+      propagateElemTypeFromInputToOutput(ctx, 0, 0);
+      if (!hasNInputShapes(ctx, 1)) {
+        return;
+      }
+
+      int64_t keep_dims = 1;
+      auto attr_proto = ctx.getAttribute("keepdims");
+      if (attr_proto) {
+        if (!attr_proto->has_i()) // incorrect attribtue
```

Comment:
I would really prefer not to silently swallow illegal error here. IMO once shape inference needs to handle error checking, the code will become very verbose, the rule should be shape inference will assume the whole model file has already passed the checks in checker.

Reply:
I agree, if onnx guarantee the model already resolved correctly, I am happy to remove those checks here.

Reply:
We'll need to return sth in these cases. I'm going to add "Status" as the return value for the inference function. Discussed with Anders before on this too. It's not added at the beginning is because there's no Status class in ONNX repo at that time.